src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
+
+  public findDuplicates(arr: number[]): number[] {
+    const duplicates: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = i + 1; j < arr.length; j++) {
+        if (arr[i] === arr[j]) {
+          duplicates.push(arr[i]);
+        }
+      }
+    }
+    return duplicates;
```

Comment:
PullProofAI

The current implementation of finding duplicates has a time complexity of O(n^2). Consider using a more efficient approach with a Set to reduce the complexity to O(n).
```diff
-     const duplicates: number[] = [];
-     for (let i = 0; i < arr.length; i++) {
-       for (let j = i + 1; j < arr.length; j++) {
-         if (arr[i] === arr[j]) {
-           duplicates.push(arr[i]);
-         }
-       }
-     }
-     return duplicates;
+     const seen = new Set<number>();
+     const duplicates = new Set<number>();
+     for (const num of arr) {
+       if (seen.has(num)) {
+         duplicates.add(num);
+       } else {
+         seen.add(num);
+       }
+     }
+     return Array.from(duplicates);
```



<!-- This is an auto-generated comment by PullProofAI -->